channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -655,7 +679,7 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                         }
                     } else if (transportType === TransportType.Fetch && isFetchSupported(syncSupport)) {
                         sendPostFunc = _fetchSender;
-                    } else if (isBeaconsSupported() && transportType === TransportType.Beacon) {
+                    } else if (transportType === TransportType.Beacon && (syncSupport && _beaconOnUnloadSupported || !syncSupport && _beaconNormalSupported)) {
```

Comment:
operator precedence may not do what you expect here, lets add some additional brackets /  rework

`((syncSupport && _beaconOnUnloadSupported) || (!syncSupport && _beaconNormalSupported))`
or
`(syncSupport ? _beaconOnUnloadSupported : _beaconNormalSupported)`